analyzer/src/main/resources/init.gradle

```diff
@@ -124,10 +124,13 @@ class DependencyTreePlugin implements Plugin<Gradle> {
             return new DependencyTreeModelImpl(project.name, configurations, repositories, errors)
         }
 
-        Dependency parseDependency(DependencyResult dependencyResult, Project project) {
+        Dependency parseDependency(DependencyResult dependencyResult, Project project, List<String> parents) {
             if (dependencyResult instanceof ResolvedDependencyResult) {
-                List<Dependency> dependencies = dependencyResult.selected.dependencies.collect { dependency ->
-                    parseDependency(dependency, project)
+                List<Dependency> dependencies = dependencyResult.selected.dependencies.findResults { dependency ->
+                    // Do not follow circular dependencies, these can exist for project dependencies.
+                    if (!(dependencyResult.requested.displayName in parents)) {
+                        parseDependency(dependency, project, [*parents, dependencyResult.requested.displayName])
+                    } else null
```

Comment:
Nit: Still put the `null` onliner into a `{ ... }` block.

Reply:
Done.